contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -2003,6 +2020,47 @@ tsql_select_common_typmod_hook(ParseState *pstate, List *exprs, Oid common_type)
 	return max_typmods;
 }
 
+/* 
+ * For CASE expression, this function will calculate typmod from tsql_select_common_typmod_hook(),
+ * and after that we will set this typmod to all the CASE branches from coerce_to_target_type().
+ */
+static Node*
+tsql_set_common_typemod_case_expr_hook(ParseState *pstate, List *exprs, CaseExpr *newc)
+{
+	int32       typmod;
+        ListCell   *l;
+
+        /* calculating common_typemod for case expr */
+        typmod = tsql_select_common_typmod_hook(pstate, exprs, newc->casetype);
+
+        newc->defresult = (Expr *) 
+                coerce_to_target_type(pstate,
+                                (Node *) newc->defresult, 
+                                newc->casetype, 
+                                newc->casetype, 
+                                typmod, 
+                                COERCION_ASSIGNMENT,
+                                COERCE_IMPLICIT_CAST,
+                                -1);
```

Comment:
Here we only want to set typmod of the CASE branches, why are we not using `coerce_type_typmod` function instead of `coerce_to_target_type` ?

Reply:
coerce_type_typmod is a static function so it couldn't be invoked in some other file.

Reply:
Removed coerce_type_typmod function as static and used this in place of coerce_to_target_type.